### ISSUE TITLE
OpenACC (GCC) and C++ AMP (Visual Studio) fallback?! 

### ISSUE THREAD

**Issue by acterhd** [amd/triSYCL#8]
Hello. Can you make fallback by GCC OpenACC and Microsoft C++ AMP fallbacks? Yes, need will write "restrict(amp, cpu)". But I'm ready for this limitation.

**Comment by keryell** [comment]
Yes, we can do what you propose with OpenACC, C++AMP or even OpenMP 4.
Right now I am more oriented towards a native implementation on top of an OpenCL 2.x Clang/LLVM implementation, to have the interoperability with the OpenCL world.
But if you have some time or resources to work on alternative approaches, I will be delighted to merge your commits.

Thank you for your proposal.

**Comment by acterhd** [comment]
OpenCL 2.1? Out not soon.

**Comment by keryell** [comment]
What do you mean by "Will not soon"?

**Comment by acterhd** [comment]
OpenCL 2.1 is out. But will not soon by AMD (probably will in new GPU). Nvidia planned support OpenCL 2.1.

**Comment by acterhd** [comment]
Again :)
http://gpuopen.com/compute-product/hcc-heterogeneous-compute-compiler/

**Comment by keryell** [comment]
Thanks for the link.

But I already added a reference to hcc in https://github.com/amd/triSYCL/commit/69281129ce734fa149dfae4bee4861870dad09bf when I updated the README,rst with related projects and the recent SYCL presentations 12 days before your message. :-)

But for now I prefer to target our resources on a more standard solution that can be upstreamed to Clang/LLVM instead of relying too much on third party environments...
